pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -34,9 +35,65 @@ const (
 	testObservationDomainID = 0xabcd
 )
 
-// TODO: We will add another test for sendDataRecord when we support adding multiple records to single set.
 // Currently, we are supporting adding only one record from one flow key to the set.
 
+func TestFlowAggregator_sendFlowKeyRecord(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockIPFIXExpProc := ipfixtest.NewMockIPFIXExportingProcess(ctrl)
+	mockIPFIXRegistry := ipfixtest.NewMockIPFIXRegistry(ctrl)
+	mockDataSet := ipfixtest.NewMockIPFIXSet(ctrl)
+	mockRecord := ipfixtest.NewMockIPFIXRecord(ctrl)
+	mockAggregationProcess := ipfixtest.NewMockIPFIXAggregationProcess(ctrl)
+
+	fa := &flowAggregator{
+		"",
+		"",
+		"tcp",
+		nil,
+		mockAggregationProcess,
+		testExportInterval,
+		mockIPFIXExpProc,
+		testTemplateIDv4,
+		testTemplateIDv6,
+		mockIPFIXRegistry,
+		mockDataSet,
+		"",
+		nil,
+		testObservationDomainID,
+	}
+
+	key1 := ipfixintermediate.FlowKey{"10.0.0.1", "10.0.0.2", 6, 1234, 5678}
+	key2 := ipfixintermediate.FlowKey{"2001:0:3238:dfe1:63::fefb", "2001:0:3238:dfe1:63::fefc", 6, 1234, 5678}
+	record1 := ipfixintermediate.AggregationFlowRecord{mockRecord, true, true}
+	record2 := ipfixintermediate.AggregationFlowRecord{mockRecord, false, true}
+
+	for key_idx, key := range []ipfixintermediate.FlowKey{key1, key2} {
```

Comment:
It would be good to put this test case struct and use that struct rather than using index which is arbitrary.
Please see the test cases in `pkg/util/flowexport`.

Reply:
Have changed to struct. Thanks! Learned a lot.